cmd/object-handlers.go

```diff
@@ -1012,8 +1012,11 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
 		return
 	}
-
-	if tags != "" {
+	if !IsStringEqual(tags, "") && !objectAPI.IsObjectTaggingSupported() {
```

Comment:
The `IsObjectTaggingSupported` can be brought inside the check for tag being empty